### PR TITLE
Feature finegrained speech control

### PR DESCRIPTION
**Pull request by tugdil**

### CODE REVIEW

**Comment by DivineThreepwood** [comment]
Hey Matthias(@tugdil), 
sorry for my late response. In order to improve the openbase pull request procedure, I created some pull request and java guidelines at https://openbase.org/guidelines/#code-guidelines

Could you revise your code regarding this guidelines again? Once you are finished just leave a comment and I will take over the revise. Feel free to ask any questions here or directly via discord: https://discord.gg/9Zhd2j2

**Comment by tugdil** [comment]
reviewed the code and removed some artefacts regarding the old version of the speech recognizer (grammar tree).
for the code guidelines (https://openbase.org/guidelines/#code-guidelines) it looks good as far as i've seen.
Only in KeywordConverter.java the function getActionParameter can return null but that was already the case before my changes.

**Comment by DivineThreepwood** [comment]
Thanks, I will finish my revise end of the week.

**Review by DivineThreepwood — CHANGES_REQUESTED** [review]
Thanks for the PR :) Code looks pretty good.  There are just some minor changes requested.  
Just be aware, there are still two commits left on your forked master branch that are not part of the PR (e25b6da, 03d3ecf).

Cheers!
Marian

File: .idea/misc.xml [hunk 1]
```diff
@@ -11,4 +11,4 @@
   <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" project-jdk-name="1.8" project-jdk-type="JavaSDK">
     <output url="file://$PROJECT_DIR$/out" />
   </component>
-</project>
```
[the remark this answers is not in the document]
Comment (tugdil) [reply .idea/misc.xml:14]: I don't understand this one. The </project> tag is still there in .idea/misc.xml

[the remark this answers is not in the document]
Comment (DivineThreepwood) [reply .idea/misc.xml:14]: I do not complain about any missing tag but about the white space modification unrelated to the pr. 
-> checkout 1.1.4 in: https://openbase.org/guidelines/#_1-contribution-guidelines

[the remark this answers is not in the document]
Comment (DivineThreepwood) [reply .idea/misc.xml:14]: you just need to perform `git checkout experimental .idea/misc.xm` in your pr branch.

File: speech/src/main/java/org/openbase/bco/psc/speech/SpeechController.java [hunk 2]
```diff
@@ -86,14 +96,69 @@ public void handleEvent(final Event event) {
...
             }
+        }
+        if (unitConfigs.size() == 0 && unitTypes.size() == 0) {
+            sendActionParameter(actionParameter);
+        }
+
+
+    }
 
+    /***
```
[the remark this answers is not in the document]
Comment (DivineThreepwood) [reply speech/src/main/java/org/openbase/bco/psc/speech/SpeechController.java:156]: It's still there right?

**Review by DivineThreepwood — CHANGES_REQUESTED** [review]

**Comment by DivineThreepwood** [comment]
Nice, just replace the `e.printStackTrace();`with the `ExceptionPrinter` usage and we are ready to go :)

**Review by DivineThreepwood — APPROVED** [review]

**Comment by DivineThreepwood** [comment]
@tugdil You just forgot to add some Imports. Next time please check if your code compiles before committing your changes :p However, I fixed all syntax errors and cleaned up some further code sections. PR is therefore ready to be merged.

**Pull request merged**